libraries/plugins/chain/chain_plugin.cpp

```diff
@@ -338,38 +335,38 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
 
    // TODO: temporary solution. DELETE when initminer mining public key is read from get_config
    my->initMiningPubkeyStr = SOPHIATX_INIT_PUBLIC_KEY_STR;
-#ifdef PRIVATE_NET
-   if (options.count("initminer-mining-pubkey")) {
+   bool private_net = false;
+   if (options.count("initminer-mining-pubkey") ) {
       my->initMiningPubkeyStr = options.at( "initminer-mining-pubkey" ).as< std::string >();
+      private_net = true;
    }
-   else {
-      BOOST_THROW_EXCEPTION( std::runtime_error("To start private net, \"initminer-mining-pubkey\" parameter must be provided.") );
-   }
-#endif //PRIVATE_NET
 
 
    auto initial_state = [&] {
-#ifdef PRIVATE_NET
-        if (options.count("initminer-account-pubkey")) {
-           // Creates genesis based on provided initminer account public key
-           genesis_state_type genesis;
-           genesis.genesis_time = time_point_sec::from_iso_string("2018-01-01T08:00:00");
-           genesis.initial_balace = SOPHIATX_INIT_SUPPLY;
-           genesis.initial_public_key = public_key_type(options.at( "initminer-account-pubkey" ).as< std::string >());
-
-           fc::sha256::encoder enc;
-           fc::raw::pack( enc, genesis );
-           genesis.initial_chain_id = enc.result();
 
-           return genesis;
-        }
-        // private net might be also started with custom genesis file
-        else {
-           if (options.count("genesis-json") == 0) {
-              BOOST_THROW_EXCEPTION( std::runtime_error("To start private net, either \"initminer-account-pubkey\" or \"genesis-json\" parameter must be provided.") );
+        if( private_net){
+           if ( options.count("initminer-account-pubkey")) {
+              // Creates genesis based on provided initminer account public key
+              genesis_state_type genesis;
+              genesis.genesis_time = time_point_sec::from_iso_string("2018-01-01T08:00:00");
+              genesis.initial_balace = 0;
+              genesis.initial_public_key = public_key_type(options.at( "initminer-account-pubkey" ).as< std::string >());
+              genesis.is_private_net = true;
+
+              fc::sha256::encoder enc;
+              fc::raw::pack( enc, genesis );
+              genesis.initial_chain_id = enc.result();
+
+              return genesis;
+           }
+           // private net might be also started with custom genesis file
+           else {
+              if (options.count("genesis-json") == 0) {
+                 BOOST_THROW_EXCEPTION( std::runtime_error("To start private net, either \"initminer-account-pubkey\" or \"genesis-json\" parameter must be provided.") );
+              }
            }
```

Comment:
In case you provide "initminer-mining-pubkey" and "genesis-json"(without "initminer-account-pubkey"), it is also valid way to start private net so genesis.is_private_net should be set to true.